File: app/src/main/java/social/entourage/android/guide/GuideMapModule.java
package social.entourage.android.guide;

import dagger.Module;
import dagger.Provides;

/**
 * Module related to GuideMapActivity
 * @see GuideMapActivity
 */
@Module
final class GuideMapModule {
    private final GuideMapEntourageFragment fragment;

    public GuideMapModule(final GuideMapEntourageFragment fragment) {
        this.fragment = fragment;
    }

    @Provides
    public GuideMapEntourageFragment providesGuideMapEntourageFragment() {
        return fragment;
    }
}

File: app/src/main/java/social/entourage/android/guide/GuideMapModule.kt
package social.entourage.android.guide

import dagger.Module
import dagger.Provides

/**
 * Module related to GuideMapFragment
 * @see GuideMapFragment
 */
@Module
internal class GuideMapModule(private val fragment: GuideMapFragment) {
    @Provides
    fun providesGuideMapFragment(): GuideMapFragment {
        return fragment
    }

}